spring-integration-zookeeper/src/main/java/org/springframework/integration/zookeeper/config/CuratorFrameworkFactoryBean.java

```diff
@@ -0,0 +1,152 @@
+/*
+ * Copyright 2015 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.integration.zookeeper.config;
+
+import org.apache.curator.RetryPolicy;
+import org.apache.curator.framework.CuratorFramework;
+import org.apache.curator.framework.CuratorFrameworkFactory;
+import org.apache.curator.framework.imps.CuratorFrameworkState;
+import org.apache.curator.retry.ExponentialBackoffRetry;
+import org.apache.curator.utils.CloseableUtils;
+
+import org.springframework.beans.factory.FactoryBean;
+import org.springframework.context.SmartLifecycle;
+import org.springframework.util.Assert;
+
+/**
+ * A spring-friendly way to build a {@link CuratorFramework} and implementing {@link SmartLifecycle}.
+ *
+ * @author Gary Russell
+ *
+ */
+public class CuratorFrameworkFactoryBean implements FactoryBean<CuratorFramework>, SmartLifecycle {
+
+	private final Object lifecycleLock = new Object();
+
+	private final CuratorFramework client;
+
+	/**
+	 * @see SmartLifecycle
+	 */
+	private volatile boolean autoStartup = true;
+
+	/**
+	 * @see SmartLifecycle
+	 */
+	private volatile boolean running;
+
+	/**
+	 * @see SmartLifecycle
+	 */
+	private volatile int phase;
+
+
+	/**
+	 * Construct an instance using the supplied connection string and using a default
+	 * retry policy {@code new ExponentialBackoffRetry(1000, 3)}.
+	 * @param connectionString
```

Comment:
I understand that these new classes will be moved to the SC, but our build system doesn't pass for such a JavaDocs.
Will be fixed on merge if everything is OK